crucible/src/Lang/Crucible/CFG/Extension/Safety.hs

```diff
@@ -262,6 +263,18 @@ class HasStructuredAssertions (ext :: Type) where
                , "else " <$$> indent 2 doc2
                ])
 
+  summarizeTree :: (IsExprBuilder sym, IsExpr (SymExpr sym))
+                => proxy ext
+                -> AssertionClassifierTree ext (RegValue' sym)
+                -> String
+  summarizeTree proxyExt =
+    intercalate ", " . nub . filter (not . ("muxing" `isPrefixOf`)) .
```

Comment:
Is this the hack you are worried about?

Reply:
Yep, that's the one. There are some messages that start with `muxing` that just aren't useful in the output (and there's a TODO comment in the source noting that already). It might be that they could be made more useful, or it might be that they could somehow never be generated? That might require some refactoring, though.

Reply:
It looks like this is the one remaining piece holding up this PR... I'm inclined to say we should leave this as-is for now, maybe with a comment describing why it's a hack and what we'd like to do better.

Perhaps a more comprehensive look at how these assertion trees are used should be a separate ticket.

Reply:
I think that sounds reasonable. Something should probably change, but it's unclear exactly what that should be yet. And, for now, the `summarizeTree` function does what its single caller needs it to do.